main.cpp

```diff
@@ -1,84 +1,39 @@
-                majorColor(major), minorColor(minor)
-            {}
-            MajorColor getMajor() {
-                return majorColor;
-            }
-            MinorColor getMinor() {
-                return minorColor;
-            }
-            std::string ToString() {
-                std::string colorPairStr = MajorColorNames[majorColor];
-                colorPairStr += " ";
-                colorPairStr += MinorColorNames[minorColor];
-                return colorPairStr;
-            }
-    };
-
-    ColorPair GetColorFromPairNumber(int pairNumber) {
-        int zeroBasedPairNumber = pairNumber - 1;
-        MajorColor majorColor = 
-            (MajorColor)(zeroBasedPairNumber / numberOfMinorColors);
-        MinorColor minorColor =
-            (MinorColor)(zeroBasedPairNumber % numberOfMinorColors);
-        return ColorPair(majorColor, minorColor);
+ namespace TelCoColorCoder
+ {
+void testNumberToPair(int pairNumber,
+        TelCoColorCoder::MajorColor expectedMajor,
+        TelCoColorCoder::MinorColor expectedMinor)
+    {
+        TelCoColorCoder::ColorPair colorPair =
+            TelCoColorCoder::GetColorFromPairNumber(pairNumber);
+        std::cout << "Got pair " << colorPair.ToString() << std::endl;
+        assert(colorPair.getMajor() == expectedMajor);
+        assert(colorPair.getMinor() == expectedMinor);
     }
-    int GetPairNumberFromColor(MajorColor major, MinorColor minor) {
-        return major * numberOfMinorColors + minor + 1;
+ 
+    void testPairToNumber(
+        TelCoColorCoder::MajorColor major,
+        TelCoColorCoder::MinorColor minor,
+        int expectedPairNumber)
+    {
+        int pairNumber = TelCoColorCoder::GetPairNumberFromColor(major, minor);
+        std::cout << "Got pair number " << pairNumber << std::endl;
+        assert(pairNumber == expectedPairNumber);
     }
 }
 
-void testNumberToPair(int pairNumber,
-    TelCoColorCoder::MajorColor expectedMajor,
-    TelCoColorCoder::MinorColor expectedMinor)
-{
-    TelCoColorCoder::ColorPair colorPair =
-        TelCoColorCoder::GetColorFromPairNumber(pairNumber);
-    std::cout << "Got pair " << colorPair.ToString() << std::endl;
-    assert(colorPair.getMajor() == expectedMajor);
-    assert(colorPair.getMinor() == expectedMinor);
-}
-
-void testPairToNumber(
-    TelCoColorCoder::MajorColor major,
-    TelCoColorCoder::MinorColor minor,
-    int expectedPairNumber)
-{
-    int pairNumber = TelCoColorCoder::GetPairNumberFromColor(major, minor);
-    std::cout << "Got pair number " << pairNumber << std::endl;
-    assert(pairNumber == expectedPairNumber);
-}
-
 int main() {
     testNumberToPair(4, TelCoColorCoder::WHITE, TelCoColorCoder::BROWN);
     testNumberToPair(5, TelCoColorCoder::WHITE, TelCoColorCoder::SLATE);
 
     testPairToNumber(TelCoColorCoder::BLACK, TelCoColorCoder::ORANGE, 12);
     testPairToNumber(TelCoColorCoder::VIOLET, TelCoColorCoder::SLATE, 25);
 
+    std::cout<<TelCoColorCoder::GetColorCoding();
```

Comment:
Good decision to separate the forming of the color codes from the printing. Think of an assert on the return value, to reduce the effort of manual testing of the output.